extension/src/minn_maxn.rs

```diff
@@ -0,0 +1,1159 @@
+use pgx::{iter::SetOfIterator, *};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    aggregate_utils::in_aggregate_context,
+    flatten,
+    palloc::{Inner, Internal, InternalAsValue, ToInternal},
+    pg_type,
+    raw::bytea,
+    ron_inout_funcs,
+};
+
+use ordered_float::NotNan;
+
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NMostTransState<T: Ord> {
```

Comment:
The module is named "minn_maxn", functions are named like "min_max_trans_function", but this is named "NMost".  Can we rename this to match?  I think I like "NMost" better so maybe rename the module and functions?

Reply:
I went with `nmost` for the generic functions, the specialized types and functions all follow `min_int` or `min_by_int`.